pina/geometry/simplex.py

```diff
@@ -0,0 +1,218 @@
+        """
+        Whether the surface should be sampled or not.
+
+        :return: Whether the surface should be sampled or not
+        :rtype: bool
+        """
+
+        return self._sample_surface
+
+    def _build_cartesian(self, vertices, labels):
+        """
+        Build Cartesian border for Simplex domain to be used in sampling.
+
+        :param vertices: list of Simplex domain's vertices
+        :type vertices: list[list]
+        :return: Cartesian border for triangular domain
+        :rtype: CartesianDomain
+        """
+
+        span_dict = {}
+
+        for i, coord in enumerate(labels):
+            sorted_vertices = sorted(vertices, key=lambda vertex: vertex[i])
+
+            # respective coord bounded by the lowest and highest values
+            span_dict[coord] = [sorted_vertices[0][i], sorted_vertices[-1][i]]
+
+        return CartesianDomain(span_dict)
+
+    def is_inside(self, point, check_border=False):
+        """
+        Check if a point is inside the simplex.
+        Uses the algorithm described here: 
+        https://math.stackexchange.com/questions/1226707/how-to-check-if-point-x-in-mathbbrn-is-in-a-n-simplex
+
+        :param point: Point to be checked
+        :type point: LabelTensor
+        :param check_border: Check if the point is also on the frontier
+            of the simplex, default False.
+        :type check_border: bool
+        :return: Returning True if the point is inside, False otherwise.
+        :rtype: bool
+        """
+
+        if not all([label in self.variables for label in point.labels]):
+            raise ValueError(
+                "Point labels different from constructor"
+                f" dictionary labels. Got {point.labels},"
+                f" expected {self.variables}."
+            )
+
+        point_shift = (point.T - self._vertices_list[:, None, -1]).type(
+            torch.FloatTensor
+        )
+        lambda_ = torch.linalg.solve(self.vectors, point_shift)
+        lambda_1 = 1.0 - torch.sum(lambda_)
+        lambdas = torch.vstack([lambda_, lambda_1])
+
+        if not check_border:
+            return all(torch.gt(lambdas, 0)) and all(torch.lt(lambdas, 1))
+
+        return all(torch.ge(lambdas, 0)) and (
+            any(torch.eq(lambdas, 0)) or any(torch.eq(lambdas, 1))
+        )
+
+    def sample(self, n, mode="random", variables="all"):
+        """
+        Sample n points from Simplex domain.
+
+        :param n: Number of points to sample in the shape.
+        :type n: int
+        :param mode: Mode for sampling, defaults to 'random'.
+            Available modes include: 'random'.
+        :type mode: str, optional
+        :param variables: pinn variable to be sampled, defaults to 'all'.
+        :type variables: str or list[str], optional
+        :return: Returns LabelTensor of n sampled points
+        :rtype: LabelTensor(tensor)
+        """
+
+        if mode != "random":
+            raise ValueError("Mode can only be random")
```

Comment:
NotImplementedError, look at ellipsoid.py

Reply:
Added the NotImplementedError